File: forum/src/main/java/com/example/forum/dto/ArticleCommentDto.java
package com.example.forum.dto;

import com.example.forum.domain.Article;
import com.example.forum.domain.ArticleComment;

import java.time.LocalDateTime;

public record ArticleCommentDto(
        Long id,
        Long articleId,
        UserAccountDto userAccountDto,
        String content,
        LocalDateTime createdAt,
        String createdBy,
        LocalDateTime modifiedAt,
        String modifiedBy
) {
    // 기본 Factory Method 생성
    public static ArticleCommentDto of(
            Long id,
            Long articleId,
            UserAccountDto userAccountDto,
            String content,
            LocalDateTime createdAt,
            String createdBy,
            LocalDateTime modifiedAt,
            String modifiedBy
    ) {
        return new ArticleCommentDto(
                id,
                articleId,
                userAccountDto,
                content,
                createdAt,
                createdBy,
                modifiedAt,
                modifiedBy
        );
    }

    // Entity To Dto
    public static ArticleCommentDto from(ArticleComment entity) {
        return new ArticleCommentDto(
                entity.getId(),
                entity.getArticle().getId(),
                UserAccountDto.from(entity.getUserAccount()),
                entity.getContent(),
                entity.getCreatedAt(),
                entity.getCreatedBy(),
                entity.getModifiedAt(),
                entity.getModifiedBy()
        );
    }

    // Article to ArticleComment
    public ArticleComment toEntity(Article entity) {
        return ArticleComment.of(
                entity,
                userAccountDto.toEntity(),
                content
        );
    }
}

File: migration-to-kotlin/src/main/kotlin/com/example/practice/dto/ArticleCommentDto.kt
package com.example.practice.dto

import com.example.practice.domain.Article
import com.example.practice.domain.ArticleComment
import java.time.LocalDateTime

data class ArticleCommentDto(
    val id : Long,
    val articleId : Long,
    val userAccountDto: UserAccountDto,
    val content : String,
    val createdAt : LocalDateTime,
    val createdBy : String,
    val modifiedAt : LocalDateTime,
    val modifiedBy : String
) {
    companion object {
        internal fun from(entity : ArticleComment) : ArticleCommentDto {
            return ArticleCommentDto(
                id = entity.id,
                articleId = entity.article.id,
                userAccountDto = UserAccountDto.from(entity.userAccount),
                content = entity.content,
                createdAt = entity.createdAt!!,
                createdBy = entity.createdBy!!,
                modifiedAt = entity.modifiedAt!!,
                modifiedBy = entity.modifiedBy!!
            )
        }
    }

    internal fun toEntity(entity : Article) : ArticleComment {
        return ArticleComment(
            article = entity,
            userAccount = userAccountDto.toEntity(),
            content = content
        )
    }
}